compiler/resolution/ResolutionCandidate.cpp

```diff
@@ -55,6 +55,11 @@ bool ResolutionCandidate::isApplicable(CallInfo& info) {
     retval = isApplicableGeneric (info);
   }
 
+  if (retval && fn->retExprType != NULL && fn->retType == dtUnknown) {
+    // Should this only happen for FLAG_GENERIC not set?
```

Comment:
I think FLAG_GENERIC does not need to be checked here because by the time it's a candidate it won't be generic anymore (but rather instantiated).